src/state_machine.rs

```diff
@@ -322,6 +351,36 @@ impl StateMachine {
         }
     }
 
+    /// Query for a result, or yield: Err(NothingAvailable), Err(Disconnected).
+    #[cfg(feature = "use-mock-crust")]
+    pub fn try_step(&mut self, outbox: &mut EventBox) -> Result<(), TryRecvError> {
+        use itertools::Itertools;
+
+        if self.is_running {
+            let mut events = Vec::new();
+            while let Ok(category) = self.category_rx.try_recv() {
+                events.push(EventType::Category(category));
+            }
+
+            let timed_out_events = self.state
+                .poll()
+                .iter()
+                .map(|token| EventType::Action(Box::new(Action::Timeout(*token))))
+                .collect_vec();
+
+            self.events.extend(events);
+            self.events.extend(timed_out_events);
```

Comment:
this is mainly to achieve as what described in the task description 2.c
`pick random index in the events from category_rx to insert timed_out event for fetched token`